tests/components/binary_sensor/test_tcp.py

```diff
@@ -1,59 +1,59 @@
 """The tests for the TCP binary sensor platform."""
+import unittest
 from copy import copy
 from unittest.mock import patch, Mock
 
 from homeassistant.components.sensor import tcp
 from homeassistant.components.binary_sensor import tcp as bin_tcp
 from tests.common import get_test_home_assistant
 from tests.components.sensor import test_tcp
+from homeassistant.bootstrap import setup_component
 
 
-@patch('homeassistant.components.sensor.tcp.Sensor.update')
-def test_setup_platform_valid_config(mock_update):
-    """Should check the supplied config and call add_entities with Sensor."""
-    add_entities = Mock()
-    ret = bin_tcp.setup_platform(None, test_tcp.TEST_CONFIG, add_entities)
-    assert ret is None, "setup_platform() should return None if successful."
-    assert add_entities.called
-    assert isinstance(add_entities.call_args[0][0][0], bin_tcp.BinarySensor)
-
-
-def test_setup_platform_invalid_config():
-    """Should check the supplied config and return False if it is invalid."""
-    config = copy(test_tcp.TEST_CONFIG)
-    del config[tcp.CONF_HOST]
-    assert bin_tcp.setup_platform(None, config, None) is False
-
-
-class TestTCPBinarySensor():
+class TestTCPBinarySensor(unittest.TestCase):
     """Test the TCP Binary Sensor."""
 
-    def setup_class(cls):
+    def setup_class(self):
         """Setup things to be run when tests are started."""
-        cls.hass = get_test_home_assistant()
+        self.hass = get_test_home_assistant()
 
-    def teardown_class(cls):
+    def teardown_class(self):
         """Stop down everything that was started."""
-        cls.hass.stop()
-
-    def test_requires_additional_values(self):
-        """Should require the additional config values specified."""
+        self.hass.stop()
+
+    def test_setup_platform_valid_config(self):
+        """Check a valid configuration."""
+        assert setup_component(
```

Comment:
`setup_component` returns `True` since #3736. More details in the PR

Please wrap this in `with assert_setup_component(1):` from `tests.common`